mqtt/mqtt-broker/tests/smoke.rs

```diff
@@ -451,9 +566,181 @@ async fn overlapping_subscriptions() {
 
     client_a.shutdown().await;
     client_b.shutdown().await;
-    broker_shutdown.send(()).expect("couldn't shutdown broker");
-    broker_task
-        .await
-        .unwrap()
-        .expect("can't wait for the broker");
+}
+
+#[tokio::test]
+async fn wrong_first_packet_connection_dropped() {
+    let broker = BrokerBuilder::default()
+        .authenticator(|_| Ok(Some(AuthId::Anonymous)))
+        .authorizer(|_| Ok(true))
+        .build();
+
+    let server_handle = common::start_server(broker);
+
+    let mut client = PacketStream::open(server_handle.address()).await;
+    client.send_packet(Packet::PingReq(PingReq)).await;
+
+    assert_matches!(client.next().await, None); // None means stream closed.
```

Comment:
You don't make match here and below, so you may want to use `assert_eq!` instead